src/lib.rs

```diff
@@ -1,6 +1,8 @@
 #![cfg_attr(feature = "alloc", feature(alloc))]
 #![no_std]
-#![deny(unsafe_code, unused)]
+#![deny(unsafe_code)]
+// TODO: Change this to enable deny(unused) if IPv6 or IPv4 are enabled
+#![cfg_attr(feature = "proto-ipv4", deny(unused))]
```

Comment:
What's holding up `#![deny(unused)]` for IPv6?

Reply:
Using it in `EthernetInterface`. Things like `Packet` and `cmp` are unused at the moment when only IPv6 is enabled